jupyterlab_latex/__init__.py

```diff
@@ -75,95 +75,110 @@ class LatexConfig(Configurable):
         help='The LaTeX command to use when compiling ".tex" files.')
     bib_command = Unicode('bibtex', config=True,
         help='The BibTeX command to use when compiling ".tex" files.')
+    shell_escape = CaselessStrEnum(['restricted', 'allow', 'disallow'],
+        default_value='restricted', config=True,
+        help='Whether to allow shell escapes '+\
+        '(and by extension, arbitrary code execution). '+\
+        'Can be "restricted", for restricted shell escapes, '+\
+        '"allow", to allow all shell escapes, or "disallow", '+\
+        'to disallow all shell escapes')
 
 
 class LatexHandler(APIHandler):
     """
     A handler that runs LaTeX on the server.
     """
-    
-    
+
     def build_tex_cmd_sequence(self, tex_base_name, run_bibtex=False):
         """Builds tuples that will be used to call LaTeX shell commands.
-        
+
         Parameters
         ----------
         tex_base_name: string
-            This is the name of the tex file to be compiled, without its 
+            This is the name of the tex file to be compiled, without its
             extension.
-            
+
         returns:
             A list of tuples of strings to be passed to
             `tornado.process.Subprocess`.
-            
+
         """
         c = LatexConfig(config=self.config)
-        
-        full_latex_sequence = (
+
+        escape_flag = ''
+        if c.shell_escape == 'allow':
+            escape_flag = '-shell-escape'
+        if c.shell_escape == 'disallow':
+            escape_flag = '-no-shell-escape'
+        full_latex_sequence = [
             c.latex_command,
             "-interaction=nonstopmode",
             "-halt-on-error",
             "-file-line-error",
-            f"{tex_base_name}",
-            )
+            ]
+        if escape_flag != '':
+            full_latex_sequence.append(escape_flag)
+
+        full_latex_sequence.append(f"{tex_base_name}")
```

Comment:
I don’t think I like this… I’d prefer for us to have the literal construction occur in one place and pass in the value even if it were to be the default anyway.

Reply:
The subprocess failed if it had an empty string in it (for some unknown reason). This is a workaround for that.

Reply:
Is there a value to pass in for restricted?

Reply:
Is there a value to pass in for restricted?

Reply:
No, that is the default.